src/kzg10/data_structures.rs

```diff
@@ -35,6 +37,47 @@ impl<E: PairingEngine> PCUniversalParams for UniversalParams<E> {
     }
 }
 
+impl<E: PairingEngine> CanonicalSerialize for UniversalParams<E> {
```

Comment:
Hmm if you're doing a manual implementation, you should probably also add implementations for the other algorithms; for example serialize_unchecked and deserialize_unchecked can be much faster than their serialize counterparts.

Reply:
Got it. So I guess I will just implement it by calling the `serialize_unchecked` of the subcomponents.

A new PR will come soon.